### ISSUE TITLE
activeClassName does not work when the links are in sub-components

### ISSUE THREAD

**Issue by drcmda** [ReactTraining/react-router#3945]
main.js

```js
ReactDOM.render((
    <Router>
        <Provider store={store} >
            <App />
        </Provider>
    </Router>
), document.getElementById('app'));
```

components.js

```js
const App = () => (
    <div>
        <Nav />
        <Content />
    </div>
)

const Nav = () => (
    <nav>
        <Link to="/overview" className="item" activeClassName="active blue">Overview</Link>
        <Link to="/logs" className="item" activeClassName="active blue">Logs</Link>
    </nav>
)

const Content = () => (
    <div>
        <Match pattern="/overview" component={Overview} />
        <Match pattern="/logs" component={Logs} />
    </div>
)

const Overview = () => <div>overview</div>
const Logs = () => <div>logs</div>
```

With the new router, how are routing informations passed down the tree? 

The Links inside the Nav component change the url, but won't indicate which route is active, it doesn't seem to apply the active classes. 

The Matches don't work at all.

Both links and matches work once, at page refresh, but then no more. If everything is directly inside the Router component it works as well. Or is this because of Redux?

**Comment by timdorr** [comment]
If you have some of those components run through `connect()`, they may be getting short circuited by some of react-redux's optimizations. Make sure your `Match`'s are outside of connect.

**Comment by spicydonuts** [comment]
More discussion (still an issue but not sure what the fix is yet): #3935

**Comment by ryanflorence** [comment]
fixed in v4.0.0-alpha.4
